src/combinatorics/simple_graph/basic.lean

```diff
@@ -355,16 +355,107 @@ lemma complete_graph_is_regular [decidable_eq V] :
 by { intro v, simp }
 
 /--
-The minimum degree of all vertices
+The minimum degree of all vertices (and `0` if there are no vertices).
 -/
-def min_degree (G : simple_graph V) [nonempty V] [decidable_rel G.adj] : ℕ :=
-finset.min' (univ.image (λ (v : V), G.degree v)) (nonempty.image univ_nonempty _)
+def min_degree [decidable_rel G.adj] : ℕ :=
+option.get_or_else (univ.image (λ v, G.degree v)).min 0
 
 /--
-The maximum degree of all vertices
+There exists a vertex of minimal degree. Note the assumption of being nonempty is necessary, as
+the lemma implies there exists a vertex.
 -/
-def max_degree (G : simple_graph V) [nonempty V] [decidable_rel G.adj] : ℕ :=
-finset.max' (univ.image (λ (v : V), G.degree v)) (nonempty.image univ_nonempty _)
+lemma exists_minimal_degree_vertex [decidable_rel G.adj] [nonempty V] :
+  ∃ v, G.degree v = G.min_degree :=
+begin
+  obtain ⟨t, ht⟩ := min_of_nonempty (univ_nonempty.image (λ v, G.degree v)),
+  have ht₂ := mem_of_min ht,
+  simp only [mem_image, mem_univ, exists_prop_of_true] at ht₂,
+  rcases ht₂ with ⟨v, rfl⟩,
+  rw option.mem_def at ht,
+  refine ⟨v, _⟩,
+  rw [min_degree, ht],
+  refl
+end
+
+/-- The minimum degree in the graph is at most the degree of any particular vertex. -/
+lemma min_degree_le_degree [decidable_rel G.adj] (v : V) : G.min_degree ≤ G.degree v :=
+begin
+  obtain ⟨t, ht⟩ := finset.min_of_mem (mem_image_of_mem (λ v, G.degree v) (mem_univ v)),
+  have := finset.min_le_of_mem (mem_image_of_mem _ (mem_univ v)) ht,
+  rw option.mem_def at ht,
+  rwa [min_degree, ht, option.get_or_else_some],
+end
+
+/--
+In a nonempty graph, if `k` is at most the degree of every vertex, it is at most the minimum
+degree. Note the assumption that the graph is nonempty is necessary as long as `G.min_degree` is
+defined to be a natural.
+-/
+lemma le_min_degree_of_forall_le_degree [decidable_rel G.adj] [nonempty V] (k : ℕ)
```

Comment:
This could be an `iff`, but I suppose that would be superfluous given the more useful previous lemma.

Reply:
Agreed